core/src/main/scala/org/apache/spark/scheduler/DAGScheduler.scala

```diff
@@ -1281,10 +1281,24 @@ class DAGScheduler(
         val failedStage = stageIdToStage(task.stageId)
         val mapStage = shuffleIdToMapStage(shuffleId)
 
+        def invalidateLostFilesAndExecutor(): Unit = {
+          // Mark the map whose fetch failed as broken in the map stage
+          if (mapId != -1) {
+            mapStage.removeOutputLoc(mapId, bmAddress)
+            mapOutputTracker.unregisterMapOutput(shuffleId, mapId, bmAddress)
+          }
+
+          // TODO: mark the executor as failed only if there were lots of fetch failures on it
+          if (bmAddress != null) {
+            handleExecutorLost(bmAddress.executorId, filesLost = true, Some(task.epoch))
+          }
+        }
+
         if (failedStage.latestInfo.attemptId != task.stageAttemptId) {
           logInfo(s"Ignoring fetch failure from $task as it's from $failedStage attempt" +
             s" ${task.stageAttemptId} and there is a more recent attempt for that stage " +
             s"(attempt ID ${failedStage.latestInfo.attemptId}) running")
+          invalidateLostFilesAndExecutor()
```

Comment:
Actually I'm confused about why you need this at all.  It looks like the map output location only gets added when the task is successful -- so what map output loc. info needs to be invalidated?